src/routes/data/handlers.ts

```diff
@@ -192,8 +192,93 @@ const handleRangeRequest = (
 
     data.stream.pipe(rangeStream).pipe(res);
   } else {
-    log.warn('Multiple ranges are not yet supported');
-    res.status(416).type('text').send('Multiple ranges are not yet supported');
+    const generateBoundary = () => {
+      // This generates a 50 character boundary similar to those used by Firefox.
+      // They are optimized for boyer-moore parsing.
+      // https://github.com/rexxars/byte-range-stream/blob/98a8e06e46193afc45219b63bc2dc5d9c7f77459/src/index.js#L115-L124
+      let boundary = '--------------------------';
+      for (let i = 0; i < 24; i++) {
+        boundary += Math.floor(Math.random() * 10).toString(16);
+      }
+
+      return boundary;
+    };
+    const boundary = generateBoundary();
+    res.status(206); // Partial Content
+    res.setHeader('Content-Type', `multipart/byteranges; boundary=${boundary}`);
+    res.setHeader('Accept-Ranges', 'bytes');
+
+    if (req.method === REQUEST_METHOD_HEAD) {
+      res.end();
+      data.stream.destroy();
+      return;
+    }
+
+    const rangeStreams: { range: any; stream: PassThrough }[] = [];
```

Comment:
nit-pick: could be this type? https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/range-parser/index.d.ts#L18-L21

Reply:
Yes

Reply:
fixed here [`ff9f4d1` (#161)](https://github.com/ar-io/ar-io-node/pull/161/commits/ff9f4d11d287d46c0dbcd5d54ac84ab72ca923a7)